packages/lib/modules/tokens/approvals/useTokenApprovalSteps.tsx

```diff
@@ -58,8 +62,16 @@ export function useTokenApprovalSteps({
   const isUnwrappingNative = wethIsEth && actionType === 'Unwrapping'
 
   const _approvalAmounts = useMemo(
-    () => approvalAmounts.filter(amount => !isSameAddress(amount.address, nativeAssetAddress)),
-    [approvalAmounts]
+    () =>
+      approvalAmounts.filter(amount => {
+        // Always filter out native asset
+        if (isSameAddress(amount.address, nativeAssetAddress)) return false
+        // If wethIsEth is true, also filter out wrapped native asset
+        if (wethIsEth && isSameAddress(amount.address, getWrappedNativeAssetAddress(chain)))
+          return false
+        return true
+      }),
+    [approvalAmounts, wethIsEth, chain]
```

Comment:
weird this logic didn't already exist since for an add liquidity with native ETH to a pool composed of WETH it shouldnt ask you to approve WETH :thinking: 

if this breaks add liquidity for some reason, there are other ways to prevent an unnecessary wrapped native asset token approval

Reply:
it doesn't ask for approval [here](https://test.balancer.fi/pools/optimism/v2/0x4fd63966879300cafafbb35d157dc5229278ed2300020000000000000000002b) though?

![image](https://github.com/user-attachments/assets/cb4f2497-f389-423e-8c4e-31adfb85733e)

Reply:
so summary modal should show unwrapped native asset